python/tests/graph/test_import.py

```diff
@@ -0,0 +1,230 @@
+    sig_outputs = {'output_sig': tf.saved_model.utils.build_tensor_info(output_tensor)}
+    serving_sigdef = tf.saved_model.signature_def_utils.build_signature_def(
+        inputs=sig_inputs, outputs=sig_outputs)
+
+    # A rather contrived way to add signature def to a meta_graph
+    meta_graph_def = tf.train.export_meta_graph()
+
+    # Find the meta_graph file (there should be only one)
+    _ckpt_meta_fpaths = glob.glob('{}/*.meta'.format(tmp_dir))
+    assert len(_ckpt_meta_fpaths) == 1, \
+        'expected only one meta graph, but got {}'.format(','.join(_ckpt_meta_fpaths))
+    ckpt_meta_fpath = _ckpt_meta_fpaths[0]
+
+    # Add signature_def to the meta_graph and serialize it
+    # This will overwrite the existing meta_graph_def file
+    meta_graph_def.signature_def[_serving_sigdef_key].CopyFrom(serving_sigdef)
+    with open(ckpt_meta_fpath, mode='wb') as fout:
+        fout.write(meta_graph_def.SerializeToString())
+
+
+def _build_saved_model(session, saved_model_dir):
+    """
+    Saves a model in a file. The graph is assumed to be generated with _build_graph_novar.
+    """
+    builder = tf.saved_model.builder.SavedModelBuilder(saved_model_dir)
+    input_tensor = tfx.get_tensor(_tensor_input_name, session.graph)
+    output_tensor = tfx.get_tensor(_tensor_output_name, session.graph)
+    sig_inputs = {'input_sig': tf.saved_model.utils.build_tensor_info(input_tensor)}
+    sig_outputs = {'output_sig': tf.saved_model.utils.build_tensor_info(output_tensor)}
+    serving_sigdef = tf.saved_model.signature_def_utils.build_signature_def(
+        inputs=sig_inputs, outputs=sig_outputs)
+
+    builder.add_meta_graph_and_variables(
+        session, [_serving_tag], signature_def_map={_serving_sigdef_key: serving_sigdef})
+    builder.save()
+
+
+@contextlib.contextmanager
+def _make_temp_directory():
+    temp_dir = tempfile.mkdtemp()
+    try:
+        yield temp_dir
+    finally:
+        shutil.rmtree(temp_dir)
+
+def _build_graph_input(gin_function):
+    """
+    Makes a session and a default graph, loads the simple graph into it, and then calls
+    gin_function(session) to return the graph input object
+    """
+    graph = tf.Graph()
+    with tf.Session(graph=graph) as s, graph.as_default():
+        _build_graph()
+        return gin_function(s)
+
+
+def _build_graph_input_var(gin_function):
+    """
+    Makes a session and a default graph, loads the simple graph into it that contains a variable,
+     and then calls gin_function(session) to return the graph input object
+    """
+    graph = tf.Graph()
+    with tf.Session(graph=graph) as s, graph.as_default():
+        _build_graph_var(s)
+        return gin_function(s)
+
+
+def _build_graph():
+    """
+    Given a session (implicitly), adds nodes of computations
+
+    It takes a vector input, with vec_size columns and returns an int32 scalar.
+    """
+    x = tf.placeholder(tf.int32, shape=[_tensor_size], name=_tensor_input_name)
```

Comment:
As we have discussed earlier in a `TensorFrames` PR, we need to test graphs with float Tensors.
This is because Spark uses `double` for vector types and TensorFlow is very strict on Tensor typing.
Mismatched types will likely end up with pages of Catalyst error info and it is hard to debug.